src/metabase/native_query_analyzer.clj

```diff
@@ -138,24 +139,32 @@
                                                     [:= :f.active true]
                                                     (into [:or] (map table-query tables))]}))))
 
-(defn field-ids-for-sql
+(defn- field-ids-for-sql
   "Returns a `{:explicit #{...} :implicit #{...}}` map with field IDs that (may) be referenced in the given card's
   query. Errs on the side of optimism: i.e., it may return fields that are *not* in the query, and is unlikely to fail
   to return fields that are in the query.
 
   Explicit references are columns that are named in the query; implicit ones are from wildcards. If a field could be
   both explicit and implicit, it will *only* show up in the `:explicit` set."
+  [driver query]
+  (let [db-id        (:database query)
+        macaw-opts   (nqa.impl/macaw-options driver)
+        sql-string   (:query (nqa.sub/replace-tags query))
+        parsed-query (macaw/query->components (macaw/parsed-query sql-string) macaw-opts)
+        explicit-ids (explicit-field-ids-for-query parsed-query db-id)
+        implicit-ids (set/difference
+                      (implicit-field-ids-for-query parsed-query db-id)
+                      explicit-ids)]
+    {:explicit   explicit-ids
+     :implicit implicit-ids}))
+
+(defn field-ids-for-native
+  "Returns a `{:explicit #{...} :implicit #{...}}` map with field IDs that (may) be referenced in the given card's
+  query. Currently only support SQL-based dialects."
   [query]
-  (when (and (active?)
-             (:native query))
-    (let [db-id        (:database query)
-          sql-string   (:query (nqa.sub/replace-tags query))
-          driver       (driver.u/database->driver db-id)
-          macaw-opts   (nqa.impl/macaw-options driver)
-          parsed-query (macaw/query->components (macaw/parsed-query sql-string) macaw-opts)
-          explicit-ids (explicit-field-ids-for-query parsed-query db-id)
-          implicit-ids (set/difference
-                        (implicit-field-ids-for-query parsed-query db-id)
-                        explicit-ids)]
-      {:explicit   explicit-ids
-       :implicit implicit-ids})))
+  (when (and (active?) (:native query))
+    (let [driver (driver.u/database->driver (:database query))]
```

Comment:
This is not really related to the PR, but should we be using `metabase.lib` to interact with the query? If we were taking modularization seriously (and I know we are breaking it all over the place) we should be using `metabase.lib.native/engine` to get the `driver` for the query

Reply:
TIL, thanks for pointing this out